### PR TITLE
Add support for issuing TLS certificates

### LINKED ISSUE #52: Support issuance of TLS certificates
For our new nuts-network efforts vendors need a TLS certificate (issued under their own CA certificate) to connect to other nodes. The certificate is also used as server-certificate.

### PR DESCRIPTION
**Pull request by reinkrul**
Fixes #52 

TODO:
 
- [x] Tests

### CODE REVIEW

File: pkg/crypto.go [hunk 1]
```diff
@@ -150,16 +154,84 @@ func (client *Crypto) SignCertificate(entity types.LegalEntity, ca types.LegalEn
...
+				//  which we don't want for our TLS certificates. Taking the 'O' component for common name should work in practice,
+				//  but is pretty ugly.
+				CommonName: caCertificate.Subject.Organization[0],
+			},
+			PublicKey: publicKey,
+		}
+		certificateAsBytes, err := client.signCertificate(&csr, entity, CertificateProfile{
+			KeyUsage:     x509.KeyUsageDigitalSignature | x509.KeyUsageKeyEncipherment | x509.KeyUsageKeyAgreement,
+			ExtKeyUsage:  []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth},
+			NumDaysValid: 60,
```
Comment (woutslakhorst) [inline pkg/crypto.go]: magic num!

**Review by woutslakhorst — COMMENTED** [review]
That Qualifier thingy smells.... it's passed through several layers, just to change the filename...

**Comment by reinkrul** [comment]
> That Qualifier thingy smells.... it's passed through several layers, just to change the filename...

We need a way to distinguish different keys/certs for an entity (org/vendor). "Qualifier" is quite generic, an alternative would be creating an enum-esque structure with all types (Intermediate CA, TLS CA, TLS, etc) but that would limit extendability in the future. Although it would't be too bad to express our certificate tree in a type-safe manner in code, as to decrease the chance of bugs?

We currently just use "the organization/vendor's key", maybe we should be explicit about it anyways (which key to use)?

**Comment by woutslakhorst** [comment]
The legalEntity + qualifier now form the `KeyIdentifier`. Maybe just call it that as an interface. This will also allow us to remove the term legalEntity from this layer. Since it doesn't really matter for the crypto if the identifier is derived from an organisation identifier.

**Comment by reinkrul** [comment]
> The legalEntity + qualifier now form the `KeyIdentifier`. Maybe just call it that as an interface. This will also allow us to remove the term legalEntity from this layer. Since it doesn't really matter for the crypto if the identifier is derived from an organisation identifier.

Refactored (types.LegalEntity, types.Qualifier) parameters to types.KeyIdentifier. Got quite elaborate.

File: pkg/types/types.go [hunk 2]
```diff
@@ -42,6 +46,43 @@ type LegalEntity struct {
 	URI string
 }
 
+type KeyIdentifier interface {
```
Comment (woutslakhorst) [inline pkg/types/types.go:50]: docs on different funcs.

**Review by woutslakhorst — CHANGES_REQUESTED** [review]
Just the docs

File: pkg/types/types.go (hunk 2 above)
Comment (reinkrul) [reply pkg/types/types.go:50]: I'd expect codestyle to fail on this?

**Review by woutslakhorst — APPROVED** [review]

**Comment by woutslakhorst** [comment]
Linter has been reconfigured, it default to python and shell..... not go

**Pull request merged**